### PR TITLE
reorganize heatmap and push gis c++ unittest

### PR DESCRIPTION
**Pull request by xge-zilliz**

### CODE REVIEW

File: cpp/src/arrow/render_api.cpp [hunk 1]
```diff
@@ -54,40 +57,51 @@ get_heatmap(std::shared_ptr<arrow::Array> arr_x, std::shared_ptr<arrow::Array> a
...
-        }
-        case arrow::Type::UINT32 : {
-            auto input_c_uint32 = (uint32_t *) arr_c->data()->GetValues<uint8_t>(1);
-            output = heatmap<uint32_t >(input_x, input_y, input_c_uint32, x_length);
-            break;
+            return out_pic(heatmap<double>(input_x, input_y, input_c_double, x_length));
         }
+
         default:
             std::cout << "type error!";
```
Comment (GuoRentong) [inline cpp/src/arrow/render_api.cpp]: use log, not cout

File: cpp/src/arrow/render_api.cpp [hunk 2]
```diff
@@ -6,20 +6,7 @@ namespace zilliz {
...
-    assert(x_length == y_length);
-    assert(x_type == arrow::Type::UINT32);
-    assert(y_type == arrow::Type::UINT32);
-
-    auto input_x = (uint32_t *) arr_x->data()->GetValues<uint8_t>(1);
-    auto input_y = (uint32_t *) arr_y->data()->GetValues<uint8_t>(1);
-
-    auto output = pointmap(input_x, input_y, x_length);
-
+out_pic(std::pair<uint8_t* ,int64_t> output) {
```
Comment (GuoRentong) [inline cpp/src/arrow/render_api.cpp]: why pair?

File: cpp/src/render/2d/cuda/heatmap.cu [hunk 3]
```diff
@@ -173,14 +173,44 @@ void set_colors_gpu(float *colors,
 } //namespace render
 } //namespace
 
+#define TEMPLATE_GEN_PREFIX
```
Comment (GuoRentong) [inline cpp/src/render/2d/cuda/heatmap.cu]: why multiple '#define TEMPLATE_GEN_PREFIX'??
do you mean #ifdefine TEMPLATE_GEN_PREFIX'?

File: cpp/src/render/2d/heatmap/heatmap.h [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #pragma once
 
 #include "render/2d/general_2d.h"
```
Comment (GuoRentong) [inline cpp/src/render/2d/heatmap/heatmap.h]: we'd better use unified include style, i.e. #include "render/2d/general_2d.h" or #include "general_2d.h", I suggest the former one

File: cpp/src/arrow/render_api.cpp (hunk 2 above)
Comment (xge-zilliz) [reply cpp/src/arrow/render_api.cpp]: output png's size is required for setting arrow::array's bitmap

File: cpp/src/arrow/render_api.cpp (hunk 1 above)
Comment (xge-zilliz) [reply cpp/src/arrow/render_api.cpp]: Log related code has not been written，I add now.

**Pull request merged**